python/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK.py

```diff
@@ -0,0 +1,434 @@
+'''
+Rule Name:
+  ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK
+
+Description:
+  Checks whether HTTP to HTTPS redirection is configured on all HTTP listeners of Application Load Balancers. The rule in NON_COMPLIANT if one or more HTTP listener of an Application Load Balancer do not have HTTP to HTTPS redirection configured.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::ElasticLoadBalancingV2::LoadBalancer
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+     Given: No Application Load Balancer is present
+      Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has only HTTPS listener(s)
+      Then: Return COMPLIANT
+  Scenario: 3
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has one or more HTTP listeners configured
+       And: At least one HTTP listener rule does not have HTTP to HTTPS redirection action configured
+      Then: Return NON_COMPLIANT
+  Scenario: 4
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has one or more HTTP listeners configured
+       And: All HTTP listener rules have HTTP to HTTPS redirection action configured
+      Then: Return COMPLIANT
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ElasticLoadBalancingV2::LoadBalancer'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+    alb_client = get_client("elbv2", event)
+    all_elbv2 = get_all_elbv2(alb_client)
+    all_elbv2 = [elb for elb in all_elbv2 if elb['Type'] == "application"]
+    if not all_elbv2:
+        return build_evaluation(event['accountId'], 'NOT_APPLICABLE', event, resource_type='AWS::::Account')
+    for elb in all_elbv2:
+        alb_all_listeners = get_all_listeners(alb_client, elb['LoadBalancerArn'])
+        overall_listeners_eval = 'NON_COMPLIANT'
+        for lis in alb_all_listeners:
+            if is_https_listener(lis):
+                overall_listeners_eval = 'COMPLIANT'
+                continue
+            rules = get_all_listener_rules(alb_client, lis['ListenerArn'])
+            for rule in rules:
+                for action in rule['Actions']:
+                    if action['Type'] == 'redirect' and action['RedirectConfig']['Protocol'] == 'HTTPS':
+                        overall_listeners_eval = 'COMPLIANT'
+                        continue
+                    break
```

Comment:
This is not right. The breaks will drive unexpected behavior. 
1. The last break will always be triggered after the first listener.
2. The previous break will trigger after the first rule